src/mongocxx/test_util/client_helpers.hh

```diff
@@ -109,6 +109,11 @@ bool is_replica_set(const client& client = {uri{}, add_test_server_api()});
 ///
 bool is_sharded_cluster(const client& client = {uri{}, add_test_server_api()});
 
+///
+/// Returns the hostname of the primary in a replica set.
+///
+std::string get_primary(const client& client = {uri{}, add_test_server_api()});
+
```

Comment:
This may not be a full solution. Other tests ([example](https://github.com/mongodb/mongo-cxx-driver/blob/c7452871978c1410153bc4da2ca7ea33c145cbf3/src/mongocxx/test/collection.cpp#L170)) with a default `uri` may also connect to a non-primary.

Consider adding a more generic `get_test_uri()` method, that could model the behavior in the C driver's test [URI functions](https://github.com/mongodb/mongo-c-driver/blob/f35c380df69f97e6f6a739707cb9917ad033f900/src/libmongoc/tests/test-libmongoc.c#L1305-L1349).